swap_meet/vendor.py

```diff
@@ -1,2 +1,63 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        items_matching_categories = []
+        for item in self.inventory:
+            if item.category == category:
+                items_matching_categories.append(item)
+        return items_matching_categories
+
+    def swap_items(self, friend, my_item, their_item):
+        if my_item in self.inventory and their_item in friend.inventory:
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+            friend.inventory.append(my_item)
+            friend.inventory.remove(their_item)
+            return True
+        else:
+            return False
```

Comment:
Here again, I would tend to invert the condition to turn this into a guard clause scenario. Compound conditions can be a little tricky to invert, but we could start with:

```python
        # if it's not the case that both vendors have their needed items
        if not (my_item in self.inventory and their_item in friend.inventory):
```

or a little more cleanly as
```python
        # if either vendor doesn't have their needed item
        if my_item not in self.inventory or their_item not in friend.inventory:
```

Notice how when moving the `not` into each partial condition, the `and` became an `or`. A similar change happens in the way we describe the situations in English.

Then we can write the whole body as 
```python
        if my_item not in self.inventory or their_item not in friend.inventory:
            return False
        
        friend.inventory.append(my_item)
        self.inventory.remove(my_item)
        self.inventory.append(their_item)
        friend.inventory.remove(their_item)
        return True 
```